### ISSUE TITLE
Signature replay attack in `CSAccounting`

### ISSUE THREAD

**Issue by djanerch1** [shieldify-security/issues#12]
## Severity

Critical

## Description

This function deposits in a staking contract on behalf of the user. But what if an attacker extracts the signature parameters from the `depositStETH()` call and frontruns it with a direct `permit()`.

## Impact

In this case, the end result is harmful since the user loses the functionality that follows the `permit()`.

## Attack Scenario

1. Attacker extracts the signature from the `depositStETH()` call.
2. Attacker frontruns it with a direct `permit()`.

## Recommendations

Use the following code when calling `permit`:

```javascript
try LIDO.permit(
                from,
                address(this),
                permit.value,
                permit.deadline,
                permit.v,
                permit.r,
                permit.s
            ) {
                return;
            } catch {
                if (LIDO.allowance(_owner, _spender) >= stETHAmount) {
                    return;
                }
            }
```

**Comment by Ptodoroff** [comment]
Mind elaborating a bit more here and providing a POC ?

**Comment by djanerch1** [comment]
I try my best to explain it better. I hope it's okay.

**Comment by ShieldifyMartin** [comment]
Front running can be shown in tests (PoCs), please provide a test to proof the validity of the finding.

https://medium.com/immunefi/how-to-reproduce-a-simple-mev-attack-b38151616cb4

**Comment by anontheauditor** [comment]
It is unclear which reason the PoC reverts. The proposed mitigation is just a different form of the already implemented check. Additional info is required.

**Comment by djanerch1** [comment]
Run the test without this lines and you will see everything is okay 
```javascript
 vm.prank(stranger);
 stETH.permit(user, address(stETH), 32 ether, 0, 0, 0, 0);
```

**Comment by anontheauditor** [comment]
yes. One can unwrap permit on from the other user. This will result in allowance increase. Hance, CSM will not attempt to unwrap permit due to already existing allowance.
